src/compiler/resolutionCache.ts

```diff
@@ -713,6 +714,10 @@ namespace ts {
                     if (!isPathWithDefaultFailedLookupExtension(fileOrDirectoryPath) && !customFailedLookupPaths.has(fileOrDirectoryPath)) {
                         return false;
                     }
+                    // prevent saving an open file from over-eagerly triggering invalidation
+                    if (resolutionHost.fileIsOpen(fileOrDirectoryPath)) {
+                        return;

```

Comment:
return false;

Reply:
Whoops, bad copy and paste from a different location.
Fixed.